src/Client/GuzzleFileFetcher.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Tuf\Client;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\RequestException;
+use Tuf\Exception\DownloadSizeException;
+use Tuf\Exception\RepoFileNotFound;
+use Tuf\JsonNormalizer;
+
+/**
+ * Defines a file fetcher that uses Guzzle to read a file over HTTPS.
+ */
+class GuzzleFileFetcher implements RepoFileFetcherInterface
+{
+    /**
+     * The HTTP client.
+     *
+     * @var \GuzzleHttp\ClientInterface
+     */
+    private $client;
+
+    /**
+     * GuzzleFileFetcher constructor.
+     *
+     * @param \GuzzleHttp\ClientInterface $client
+     *   The HTTP client.
+     */
+    public function __construct(ClientInterface $client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * Creates an instance of this class with a specific base URI.
+     *
+     * @param string $baseUri
+     *   The base URI from which to fetch files.
+     *
+     * @return static
+     *   A new instance of this class.
+     */
+    public static function createFromUri(string $baseUri) : self
+    {
+        $scheme = parse_url($baseUri, PHP_URL_SCHEME);
+        if ($scheme === 'https') {
+            $client = new Client(['base_uri' => $baseUri]);
+            return new static($client);
+        } else {
+            throw new \InvalidArgumentException("Repo base URI must be HTTPS: $baseUri");
```

Comment:
Not sure if we should allow non `https`. I know @davidstrauss has mention that using TUF we don't have to trust the HTTPS.

Reply:
We don't have to, but maybe we should? It seems like a reasonable restriction, at least for now.

Reply:
ok fine leaving as is until we get a request for this(and maybe rejecting the request)